services/src/main/java/org/fao/geonet/api/records/backup/ArchiveAllMetadataJob.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.domain.Specifications;
+import org.springframework.scheduling.quartz.QuartzJobBean;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nullable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Service
+public class ArchiveAllMetadataJob extends QuartzJobBean {
+
+    @Autowired
+    private ConfigurableApplicationContext context;
+    @Autowired
+    private ServiceManager serviceManager;
+
+    @Autowired
+    private SettingManager settingManager;
+
+    static final String BACKUP_FILENAME = "gn_backup";
+    static final String BACKUP_DIR = "backup_archive";
+    public static final String BACKUP_LOG = Geonet.GEONETWORK + ".backup";
+    private AtomicBoolean backupIsRunning = new AtomicBoolean(false);
+
+
+    @Override
+    protected void executeInternal(JobExecutionContext jobContext) throws JobExecutionException {
+        ServiceContext serviceContext = serviceManager.createServiceContext("unpublishMetadata", context);
+        serviceContext.setLanguage("eng");
+        serviceContext.setAsThreadLocal();
+
+        ApplicationContextHolder.set(this.context);
+
+        if(!settingManager.getValueAsBool(Settings.METADATA_BACKUPARCHIVE_ENABLE)) {
+            Log.info(BACKUP_LOG, "Backup archive not enabled");
+            return;
+        }
+
+        try {
+            createBackup(serviceContext);
+        } catch (Exception e) {
+            Log.error(Geonet.GEONETWORK, "Error running " + ArchiveAllMetadataJob.class.getSimpleName(), e);
+        }
+    }
+
+    public void createBackup(ServiceContext serviceContext) throws Exception {
+
+        ApplicationContext appContext = ApplicationContextHolder.get();
+        GeonetworkDataDirectory dataDirectory = appContext.getBean(GeonetworkDataDirectory.class);
+
+        Path stylePath = dataDirectory.getWebappDir().resolve(Geonet.Path.SCHEMAS);
+
+        if (!backupIsRunning.compareAndSet(false, true)) {
+            return;
+        }
+        long startTime = System.currentTimeMillis();
+        try {
+            Log.info(BACKUP_LOG, "Starting backup of all metadata");
+            System.out.println("Starting backup of all metadata");
+
+            final MetadataRepository metadataRepository = serviceContext.getBean(MetadataRepository.class);
+
+            loginAsAdmin(serviceContext);
+            final Specification<Metadata> harvested = Specifications.where(MetadataSpecs.isHarvested(false)).
+                    and(Specifications.not(MetadataSpecs.hasType(MetadataType.SUB_TEMPLATE)));
```

Comment:
It would be good to explain that (ie. backup contains only non harvested records (including templates)) to the end user. I would add that to the API notes and to the Angular app translations.